sensors/src/main/scala/com/lynbrookrobotics/potassium/sensors/imu/IMURegister.scala

```diff
@@ -0,0 +1,54 @@
+package com.lynbrookrobotics.potassium.sensors.imu
+
+import com.lynbrookrobotics.potassium.sensors.SPI
+
+/**
+  * Import statements for com.lynbrookrobotics.potassium.sensors.SPI and code involving Bytes
+  */
+import java.nio.ByteBuffer
+
+/**
+  * Represents a register on the ADIS16448 IMU.
+  */
+
+/**
+  *
+  * @param register is ID of register on IMU
+  */
+case class IMURegister(register: Int) {
+  private val readBuffer: ByteBuffer = ByteBuffer.allocateDirect(2)
+  // note that casting is not supported by Scala, so must use .asInstanceOf[]
```

Comment:
Not true, `asInstanceOf` is how you cast in Scala. So casting is supported, it just happens in a different way.